File: app/src/main/java/com/paradigmadigital/dagger/ui/detail/DetailPresenter.java
package com.paradigmadigital.dagger.ui.detail;

import com.paradigmadigital.dagger.ui.ActCollaborator;
import com.paradigmadigital.dagger.ui.Navigator;

import android.util.Log;

import javax.inject.Inject;

public class DetailPresenter {

    private static final String TAG = DetailPresenter.class.getSimpleName();

    private final Navigator navigator;
    private final ActCollaborator actColaborator;

    @Inject
    public DetailPresenter(Navigator navigator, ActCollaborator actColaborator) {
        this.navigator = navigator;
        this.actColaborator = actColaborator;
    }

    public void initialize() {

    }

    public void onClick() {
        showLogs();
        navigator.navigateToAnother();
    }

    private void showLogs() {
        Log.d(TAG, "ActCollaborator: " + actColaborator);
    }
}

File: module_detail/src/main/java/eu/sesma/dagger/detail/DetailPresenter.kt
package eu.sesma.dagger.detail

import android.util.Log
import eu.sesma.dagger.core.CoreCollaborator
import eu.sesma.dagger.core.CoreSingletonCollaborator
import javax.inject.Inject

class DetailPresenter
@Inject constructor(private val navigator: DetailNavigator,
                    private val coreSingletonCollaborator: CoreSingletonCollaborator,
                    private val coreCollaborator: CoreCollaborator) {

    companion object {
        private val TAG = DetailPresenter::class.java.simpleName
    }

    fun initialize() {
    }

    fun onClick() {
        showLogs()
        navigator.navigateToAnother()
    }

    private fun showLogs() {
        Log.d(TAG, "CoreCollaborator: " + coreCollaborator +
                " \r\nCoreSingletonCollaborator: " + coreSingletonCollaborator)
    }
}
